src/test/controller/UserController.test.ts

```diff
@@ -195,52 +245,43 @@ describe('Unit tests for UserController', function () {
       });
   });
 
-  test('GET /users/:id/savedPromotions', async (done) => {
-    const expectedUser: User = new UserFactory().generate();
-    const promotion = new PromotionFactory().generate(
-      expectedUser,
-      new DiscountFactory().generate(),
-      [new ScheduleFactory().generate()]
-    );
-
-    await userRepository.save(expectedUser);
-    await promotionRepository.save(promotion);
-
-    // create saved promotions
-    await savedPromotionRepository.save(
-      new SavedPromotionFactory().generate(expectedUser, promotion)
-    );
-    request(app)
-      .get(`/users/${expectedUser.id}/savedPromotions`)
-      .expect(200)
-      .end((err, res) => {
-        if (err) return done(err);
-        const user = res.body;
-        expect(user).toHaveProperty('savedPromotions');
-        compareUsers(user, expectedUser);
-        expect(user.savedPromotions).toHaveLength(1);
-        expect(user.savedPromotions[0]).toMatchObject({
-          userId: expectedUser.id,
-          promotionId: promotion.id,
-          promotion: { id: promotion.id, name: promotion.name },
+  xtest('GET /users/:id/savedPromotions', async () => {
+    for (let i = 0; i < 100; i++) {
```

Comment:
Is there a reason we're running this 100 times? 😅

Reply:
Ah sorry this was me loool this test is gonna take a really long time. @yunishimura0716 sorry I didn't know you were going to merge in my branch `user-endpoints-deadlock`, that branch was just me playing with things to test the deadlock. Once I merge in https://github.com/ubclaunchpad/PromoPal-backend/pull/121, then your branch should have the proper code